### PR TITLE
NRPT-686 Add Map Layer Info controller

### PR DESCRIPTION
**Pull request by BcGovNeal**
https://bcmines.atlassian.net/browse/NRPT-686

Test payloads:

`POST /api/map-info`
```
{
    "application": "LNG",
    "data": {
        "location": "Nile River",
        "length": "48.0 km",
        "description": "paragraph of formatted text",
        "segment": "Section 1"
    }
}
```

`PUT /api/map-info/<id>`

```
{
    "location": "Nile River",
    "length": "48.0 km",
    "description": "paragraph of formatted text"
}
```

`DELETE /api/map-info/<id>`

`GET /api/map-info/<id>`

`GET /api/public/map-info/<id>`

### CODE REVIEW

File: api/src/controllers/map-info-controller.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
+// const mongoose = require('mongoose');
+// const ObjectID = require('mongodb').ObjectID;
```
Comment (mtCarto) [inline api/src/controllers/map-info-controller.js:2]: Remove since they aren't used

**Review by mtCarto — APPROVED** [review]
Looks good, CRUD operations are working correctly

File: api/src/controllers/map-info-controller.js (hunk 1 above)
Comment (BcGovNeal) [reply api/src/controllers/map-info-controller.js]: Fixed in https://github.com/bcgov/NRPTI/pull/790/commits/e14e75a6eafb62e6935dd5b1f8639c3561a8886c

File: api/src/controllers/map-info-controller.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+};
+
+async function getMapLayerInfoSegment(segment) {
+  return await MapLayerInfo.findOne({ _schemaName: 'MapLayerInfo', segment });
+}
+
+exports.protectedPost = async function(args, res, next) {
+  let incomingObj = {};
+  let errorMsg = null;
+  // Only accpet applicaiton: 'LNG' for now
```
Comment (marklise) [inline api/src/controllers/map-info-controller.js:82]: sp.

Comment (BcGovNeal) [reply api/src/controllers/map-info-controller.js]: Fix in https://github.com/bcgov/NRPTI/pull/790/commits/aa0dec560bb5f665fcf3ba1a7c25037c5b122711

**Review by marklise — APPROVED** [review]

**Pull request merged**